src/number_theory/pell.lean

```diff
@@ -398,4 +427,252 @@ end solution₁
+lemma pow_eq_one_iff {a : solution₁ d} (h : is_fundamental a) (n : ℤ) : a ^ n = 1 ↔ n = 0 :=
+begin
+  refine ⟨λ H, _, λ H, by rw [H, zpow_zero]⟩,
+  have H' : (a ^ n).y = 0 := by rw [H, y_one],
+  rcases lt_trichotomy 0 n with hn | rfl | hn,
+  { exact absurd H' (y_zpow_pos h.x_pos h.2.1 hn).ne', },
+  { refl, },
+  { have := y_zpow_pos h.x_pos h.2.1 (by rwa [lt_neg, neg_zero] : 0 < -n),
+    rw [zpow_neg, y_inv, H', neg_zero] at this,
+    exact false.elim (lt_irrefl _ this), }
+end
+
+/-- The `n`th power of a fundamental solution has positive `y` if and only if `n` is positive. -/
+lemma pow_y_pos_iff {a : solution₁ d} (h : is_fundamental a) (n : ℤ) : 0 < (a ^ n).y ↔ 0 < n :=
+begin
+  refine ⟨λ H, _, y_zpow_pos h.x_pos h.2.1⟩,
+  rcases lt_trichotomy 0 n with hn | rfl | hn,
+  { exact hn, },
+  { rwa [zpow_zero, y_one] at H, },
+  { have := y_zpow_pos h.x_pos h.2.1 (by rwa [lt_neg, neg_zero] : 0 < -n),
+    rw [zpow_neg, y_inv, lt_neg, neg_zero] at this,
+    exact false.elim (lt_irrefl _ $ H.trans this), }
+end
+
+/-- The `n`th power of a fundamental solution has negative `y` if and only if `n` is negative. -/
+lemma pow_y_neg_iff {a : solution₁ d} (h : is_fundamental a) (n : ℤ) : (a ^ n).y < 0 ↔ n < 0 :=
+begin
+  rw [← neg_neg n, zpow_neg, y_inv, neg_lt, neg_zero, neg_lt, neg_zero],
+  exact h.pow_y_pos_iff (-n),
+end
+
+/-- A power of a fundamental solution is never equal to the negative of a power of this
+fundamental solution. -/
+lemma pow_ne_neg_pow {a : solution₁ d} (h : is_fundamental a) {n n' : ℤ} :
+   a ^ n ≠ -a ^ n' :=
+begin
+  intro hf,
+  apply_fun solution₁.x at hf,
+  have H := x_zpow_pos h.x_pos n,
+  rw [hf, x_neg, lt_neg, neg_zero] at H,
+  exact lt_irrefl _ ((x_zpow_pos h.x_pos n').trans H),
+end
+
+/-- The `x`-coordinate of a fundamental solution is a lower bound for the `x`-coordinate
+of any positive solution. -/
+lemma x_le_x {a₁ : solution₁ d} (h : is_fundamental a₁) {a : solution₁ d} (hax : 1 < a.x) :
+  a₁.x ≤ a.x :=
+h.2.2 hax
+
+/-- The `y`-coordinate of a fundamental solution is a lower bound for the `y`-coordinate
+of any positive solution. -/
+lemma y_le_y {a₁ : solution₁ d} (h : is_fundamental a₁) {a : solution₁ d} (hax : 1 < a.x)
+  (hay : 0 < a.y) :
+  a₁.y ≤ a.y :=
+begin
+  have H : d * (a₁.y ^ 2 - a.y ^ 2) = a₁.x ^ 2 - a.x ^ 2 := by { rw [a.prop_x, a₁.prop_x], ring },
+  rw [← abs_of_pos hay, ← abs_of_pos h.2.1, ← sq_le_sq, ← mul_le_mul_left h.d_pos, ← sub_nonpos,
+      ← mul_sub, H, sub_nonpos, sq_le_sq, abs_of_pos (zero_lt_one.trans h.1),
+      abs_of_pos (zero_lt_one.trans hax)],
+  exact h.x_le_x hax,
+end
+
+/-- If we multiply a positive solution with the inverse of a fundamental solution,
+the `x`-coordinate decreases and the `y`-coordinate remains nonnegative. -/
+lemma x_mul_inv_lt_x {a₁ : solution₁ d} (h : is_fundamental a₁) {a : solution₁ d} (hax : 1 < a.x)
+  (hay : 0 < a.y) :
+  0 < (a * a₁⁻¹).x ∧ (a * a₁⁻¹).x < a.x ∧ 0 ≤ (a * a₁⁻¹).y :=
```

Comment:
I think this should be three separate lemmas, rather than having an `and` in the conclusion.

Reply:
OK; it is now split into four parts (since the statement called `H₁` in the original proof (now `x_mul_y_le_y_mul_x`) is used in all three parts of the original conjunction).